File: parserTWD_Bloomberg.pl
my $s;

open(HTM, $ARGV[0]) or die "$!";
while (defined($line = <HTM>)) {
	$s = $s.$line;
}
close(HTM);

my $time = "quoteTime";
my $range = index($s, $time);
#my $date = substr($s, $range + 20, 10);
#print "date=".($date)."\n";

my $year = substr($s, $range + 20, 4);
my $mm = substr($s, $range + 25, 2);
my $dd = substr($s, $range + 28, 2);

print "date=".($year).($mm).($dd)."\n";
print "type=bloomberg_USDTWD\n";

my $last = "Previous Close:";
my $range = index($s, $last);

print "price=".substr($s, $range + 35, 7)."\n";

File: parserTWD_Bloomberg.py
# coding=UTF-8
import sys
import sqlite3

filename = str(sys.argv[1])
 
f = open(filename, 'r')
b_str = f.read()
f.close()

index = b_str.find('quoteTime') + 20
yy = b_str[index:]
yy = yy[:4]
index += 5
mm = b_str[index:]
mm = mm[:2]
index += 3
dd = b_str[index:]
dd = dd[:2]

date = yy + '-' + mm + '-' + dd

print 'date=' + date
print 'type=bloomberg_USDTWD'

index = b_str.find('Previous Close:') + 35
price = b_str[index:]
price = price[:7]
print "price=" + price;

conn = sqlite3.connect('oil_db.sqlite')
c = conn.cursor()
c.execute('SELECT * FROM stocks WHERE time=? and type=?', (date, "bloomberg_USDTWD"))
result = c.fetchone()

if result is None:
	c.execute('INSERT INTO stocks VALUES (?, ?, ?)', (date, "bloomberg_USDTWD", float(price)))
	conn.commit()
else:
	print date + ' bloomberg_USDTWD exist'

c.close()
